CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(d3d8to9 CXX)
+
+set(SRCS
+  source/d3d8to9_base.cpp
+  source/d3d8to9.cpp
+  source/d3d8to9_device.cpp
+  source/d3d8to9.hpp
+  source/d3d8to9_index_buffer.cpp
+  source/d3d8to9_surface.cpp
+  source/d3d8to9_swap_chain.cpp
+  source/d3d8to9_texture.cpp
+  source/d3d8to9_vertex_buffer.cpp
+  source/d3d8to9_volume.cpp
+  source/d3d8types.cpp
+  source/d3d8types.hpp
+  source/d3dx9.hpp
+  source/interface_query.cpp
+  source/interface_query.hpp
+)
+
+set(CMAKE_CXX_STANDARD 17)
+
+option(D3D8TO9_STATIC "Build as a static library" OFF)
+
+if(D3D8TO9_STATIC)
+  add_library(d3d8to9 STATIC ${SRCS})
+else()
+  add_library(d3d8to9 SHARED ${SRCS})
+  set_target_properties(d3d8to9 PROPERTIES PREFIX "")
+  set_target_properties(d3d8to9 PROPERTIES OUTPUT_NAME "d3d8")
+endif()
+
+target_compile_definitions(d3d8to9 PUBLIC
+  $<$<CONFIG:Debug>:D3D_DEBUG_INFO;_DEBUG;aaa>
```

Comment:
What's up with the `aaa`? I'm guessing this is left-over debugging code?
Btw., doesn't CMake already set `NDEBUG` automatically, so it should not be necessary to include that here? On that note I see there is some use of `_DEBUG` in d3d8to9 which should be changed to `NDEBUG` at some point ... but that's not necessarily in scope for this PR.

Reply:
Fixed! I don't know how could I forget about that 'aaa' :P

Reply:
I've seen you readded `_DEBUG` definition back in 654588e408a985e2474cc8ca82e27f7854b82530 . Is there any reason for that? You also didn't make analogous change in CMakeBuild.txt

Reply:
As it was CMake did not add the `NDEBUG` definition to the resource compiler settings, nor was it set in the Visual Studio project. As a result the DLL was always build with `VS_FF_DEBUG` file flag, which was incorrect. To simplify things I chose to use `_DEBUG` in this case again, so that by default it always builds without that file flag and only if `_DEBUG` is eplicitly set it sets `VS_FF_DEBUG`. With CMake this is not set, but that's fine.